plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -847,9 +851,11 @@ public Optional<ConnectorOutputMetadata> finishInsert(ConnectorSession session,
 
         IcebergWritableTableHandle table = (IcebergWritableTableHandle) insertHandle;
         Table icebergTable = transaction.table();
+        Optional<Long> previousSnapshotId = Optional.ofNullable(icebergTable.currentSnapshot()).map(Snapshot::snapshotId);
```

Comment:
This looks more like current not previous.

Reply:
it's "previous" from the point of view of the ongoing write 
changed to `beforeWriteSnapshotId`